File: Practica 1/Punto.py
import math

class Punto:
    def __init__(self, x, y):
        self.x = float(x)
        self.y = float(y)

    def coord_cartesianas(self):
        return self.x, self.y

    def coord_polares(self):
        radio = math.sqrt(self.x ** 2 + self.y ** 2)
        angulo = math.atan2(self.y, self.x) 
        angulo = math.degrees(angulo)
        return radio, angulo

    def __str__(self):
        return "({:.2f},{:.2f})".format(self.x, self.y)

# Ejemplo de uso
p1 = Punto(1, 5)

print(p1)
x, y = p1.coord_cartesianas()
print(x, y)
r, a = p1.coord_polares()
print(r, a)

File: Practica 1/Punto.java
public class Punto {
    public float x;
    public float y;

    public Punto(float x, float y) {
        this.x = x;
        this.y = y;
    }

    public float[] coordCartesianas() {
        return new float[] { x, y };
    }

    public float[] coordPolares() {
        float radio = (float) Math.sqrt(x * x + y * y);
        float angulo = (float) Math.toDegrees(Math.atan2(y, x)); // Usar atan2 para obtener el ángulo correcto
        return new float[] { radio, angulo };
    }

    @Override
    public String toString() {
        return String.format("(%.2f, %.2f)", x, y);
    }

    public static void main(String[] args) {
        Punto p1 = new Punto(1, 5);

        float[] cartesianas = p1.coordCartesianas();
        float[] polares = p1.coordPolares();
    }
}